File: VB/WindowsApplication1/CustomEditor/MyLookUpEdit.vb
Imports System
Imports System.ComponentModel
Imports DevExpress.XtraEditors

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory("")>
	Public Class MyLookUpEdit
		Inherits LookUpEdit

		Shared Sub New()
			RepositoryItemMyLookUpEdit.Register()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMyLookUpEdit.EditorName
			End Get
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
		Public Shadows ReadOnly Property Properties() As RepositoryItemMyLookUpEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMyLookUpEdit)
			End Get
		End Property


		Protected Overrides Sub ProcessFindItem(ByVal helper As DevExpress.XtraEditors.Controls.KeyPressHelper, ByVal pressedKey As Char)
			AssignDataSource(AutoSearchText)
			MyBase.ProcessFindItem(helper, pressedKey)
		End Sub

		Private Sub AssignDataSource(ByVal autoSearchText As String)
			Properties.UpdateDataSource(autoSearchText)
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/CustomEditor/MyLookUpEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyLookUpEdit : LookUpEdit
    {
        static MyLookUpEdit()
        {
            RepositoryItemMyLookUpEdit.Register();
        }
     
        public override string EditorTypeName
        {
            get { return RepositoryItemMyLookUpEdit.EditorName; }
        }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemMyLookUpEdit Properties
        {
            get { return base.Properties as RepositoryItemMyLookUpEdit; }
        }


        protected override void ProcessFindItem(DevExpress.XtraEditors.Controls.KeyPressHelper helper, char pressedKey)
        {
            AssignDataSource(AutoSearchText);
            base.ProcessFindItem(helper, pressedKey);
        }

        private void AssignDataSource(string autoSearchText)
        {
            Properties.UpdateDataSource(autoSearchText);
        
        }

    }
}
